src/node_ffi.cc

```diff
@@ -0,0 +1,283 @@
+#include "node_ffi.h"
+#include "env-inl.h"
+#include "ffi.h"
+#include "node_binding.h"
+#include "node_external_reference.h"
+#include "util-inl.h"
+#include "v8-array-buffer.h"
+#include "v8-fast-api-calls.h"
+#include "v8-function-callback.h"
+#include "v8-primitive.h"
+#include "v8-template.h"
+
+using v8::Array;
+using v8::ArrayBuffer;
+using v8::BigInt;
+using v8::Boolean;
+using v8::Context;
+using v8::FunctionCallbackInfo;
+using v8::FunctionTemplate;
+using v8::Isolate;
+using v8::Number;
+
+namespace node {
+namespace ffi {
+
+static bool lossless = false;
+
+#define ReturnBigInt(isolate, ptr)                                             \
+  args.GetReturnValue().Set(BigInt::NewFromUnsigned(isolate, (uint64_t)ptr))
+
+void GetLibrary(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  THROW_IF_INSUFFICIENT_PERMISSIONS(env, permission::PermissionScope::kFfi, "");
+
+  bool isNull = true;
+  std::string fname = "";
+  if (!args[0]->IsNull()) {
+    CHECK(args[0]->IsString());
+    node::Utf8Value filename(env->isolate(), args[0]);
+    fname = filename.ToString();
+    isNull = false;
+  }
```

Comment:
Could simplify that to something like:

```js
  std::string fname = "";
  CHECK(args[0]->IsNull() || args[0]->IsString());
  if (args[0]->IsString()) {
    fname = *node::Utf8Value(env->isolate(), args[0]);
  }
```

Then you can check `fname.empty()` rather than needing `isNull`.